File: src/main/java/com/sfh/pokeRogueBot/model/browser/pokemonjson/TurnData.java
package com.sfh.pokeRogueBot.model.browser.pokemonjson;

import lombok.Data;

import java.util.List;

@Data
public class TurnData {

    private int damageDealt;
    private int currDamageDealt;
    private int damageTaken;
    private List<AttacksReceived> attacksReceived;
    private boolean acted;
    private int hitCount;
    private int hitsLeft;
}

File: src/main/kotlin/com/sfh/pokeRogueBot/model/browser/pokemonjson/TurnData.kt
package com.sfh.pokeRogueBot.model.browser.pokemonjson

data class TurnData(
    val damageDealt: Int,
    val currDamageDealt: Int,
    val damageTaken: Int,
    val attacksReceived: List<AttacksReceived>?,
    val acted: Boolean,
    val hitCount: Int,
    val hitsLeft: Int
)